lua/nvim_lsp/codeqlls.lua

```diff
@@ -0,0 +1,46 @@
+local configs = require "nvim_lsp/configs"
+local util = require "nvim_lsp/util"
+
+local server_name = "codeqlls"
+
+local root_pattern = util.root_pattern("qlpack.yml")
+
+configs[server_name] = {
+    default_config = {
+        filetypes = {'codeql'};
+        root_dir = function(fname)
+            return root_pattern("qlpack.yml") or util.path.dirname(fname)
+        end;
+        log_level = vim.lsp.protocol.MessageType.Warning;
+        before_init = function(initialize_params, config)
+            initialize_params['workspaceFolders'] = {{
+                    name = 'workspace',
+                    uri = initialize_params['rootUri']
+            }}
+        end;
+        settings = {
+            search_path = vim.empty_dict()
+        };
+    };
+    docs = {
+        description = [[
+Reference:
+https://help.semmle.com/codeql/codeql-cli.html
+
+Binaries:
+https://github.com/github/codeql-cli-binaries
```

Comment:
I think we should also mention how to use on_new_config

Reply:
Sure will add